### PR TITLE
feat: default select menu values

### PR DESCRIPTION
**Pull request by jaw0r3k**
**Please describe the changes this PR makes and why it should be merged:**
Q: Should discord.js also allow to pass structures (overwrite that like in case of setting embed colors)?

Adds:
- https://github.com/discord/discord-api-docs/pull/6442
- https://github.com/discordjs/discord-api-types/pull/824

Adds methods to every auto populated select builder like
- `addDefaultUsers`
- or `setDefaultRoles`

Explicitly Mentionable Select has 
setDefaultValues as it supports roles and users

**Status and versioning classification:**

- Code changes have been tested against the Discord API, or there are no code changes
- I know how to update typings and have done so, or typings don't need updating
- This PR changes the library's interface (methods or parameters added)

### CODE REVIEW

**Review by Jiralite — COMMENTED** [review]
Instead of adding the same two methods to four classes, perhaps a base automatically-populated select menu class should be made for them to inherit from? So, those named methods could be called `setDefaultValues()`, as that's the property they are updating.

**Comment by jaw0r3k** [comment]
And then force users to use full objects?

**Comment by jaw0r3k** [comment]
A bit complicated as ts confuses to merge types together as they are split on api types
image

File: packages/discord.js/typings/index.d.ts [hunk 1]
```diff
@@ -586,9 +586,6 @@ export abstract class CommandInteraction<Cached extends CacheType = CacheType> e
     option: APIApplicationCommandOption,
     resolved: APIApplicationCommandInteractionData['resolved'],
   ): CommandInteractionOption<Cached>;
-  private transformResolved(
```
Comment (vladfrangu) [inline packages/discord.js/typings/index.d.ts:589]: lets export this method still, as it was technically exported before

**Review by vladfrangu — CHANGES_REQUESTED** [review]

File: packages/discord.js/typings/index.d.ts (hunk 1 above)
Comment (Jiralite) [reply packages/discord.js/typings/index.d.ts:589]: Resolved with 420288d90e94f218f780323bd85930d20b6132e3.

**Review by vladfrangu — APPROVED** [review]

**Review by SpaceEEC — APPROVED** [review]

File: packages/builders/src/components/selectMenu/MentionableSelectMenu.ts [hunk 2]
```diff
@@ -31,4 +38,55 @@ export class MentionableSelectMenuBuilder extends BaseSelectMenuBuilder<APIMenti
...
+		this.data.default_values.push(...normalizedValues);
+		return this;
+	}
+
+	/**
+	 * Sets default values to this auto populated select menu.
+	 *
+	 * @param values - The values to set
+	 */
+	public setDefaultValues(
```
Comment (kyranet) [inline packages/builders/src/components/selectMenu/MentionableSelectMenu.ts:81]: Why is there no `addDefaultValues`?

File: packages/builders/src/components/selectMenu/ChannelSelectMenu.ts [hunk 3]
```diff
@@ -59,6 +64,39 @@ export class ChannelSelectMenuBuilder extends BaseSelectMenuBuilder<APIChannelSe
...
+	 */
+	public addDefaultChannels(...channels: RestOrArray<Snowflake>) {
+		const normalizedValues = normalizeArray(channels).map((id) => ({
+			id,
+			type: SelectMenuDefaultValueType.Channel as const,
+		}));
+
+		this.data.default_values ??= [];
+		optionsLengthValidator.parse(this.data.default_values.length + normalizedValues.length);
+		this.data.default_values.push(...normalizedValues);
```
Comment (kyranet) [inline packages/builders/src/components/selectMenu/ChannelSelectMenu.ts:80]: Changing the operation order to not map or ensure `default_values` will reduce overhead in the case of the validator throwing an error.

- normalizeArray → validate length → ensure array → map values → push to array

**Review by kyranet — DISMISSED** [review]

File: packages/builders/src/components/selectMenu/MentionableSelectMenu.ts (hunk 2 above)
Comment (Jiralite) [reply packages/builders/src/components/selectMenu/MentionableSelectMenu.ts:99]: Resolved with 1126cd91c4fe9d7910a364a593913de7d87b78fe.

File: packages/discord.js/typings/index.d.ts [hunk 4]
```diff
@@ -6363,6 +6365,12 @@ export interface StageInstanceEditOptions {
   privacyLevel?: StageInstancePrivacyLevel;
 }
 
+export interface SupportingInteractionResolvedData {
+  client: Client;
+  guild?: Guild;
+  channel?: Channel;
```
Comment (vladfrangu) [inline packages/discord.js/typings/index.d.ts:6371]: it needs to be a channel with messages in it

**Review by vladfrangu — CHANGES_REQUESTED** [review]

File: packages/builders/src/components/selectMenu/ChannelSelectMenu.ts (hunk 3 above)
Comment (Jiralite) [reply packages/builders/src/components/selectMenu/ChannelSelectMenu.ts]: Should be done with fa4833ceec160487b6ea423548fdac317874f7f1.

File: packages/discord.js/typings/index.d.ts (hunk 4 above)
Comment (Jiralite) [reply packages/discord.js/typings/index.d.ts]: Done with de41e9540f9cead1708dd1ed2aab741a6a132aef.

**Review by vladfrangu — APPROVED** [review]

**Review by kyranet — APPROVED** [review]

**Pull request merged**